### ISSUE TITLE
Prevent accidental opening link

### ISSUE THREAD

**Issue by anedroid** [maltaisn/another-notes-app#108]
**The problem**
When editing a note, it's super easy to accidentally touch on the link and it opens. Most of the times however I touch there not because I want to open it, but in order to set cursor there, append comma or return and put another link. The actual behaviour makes poor UX.

**The solution**
I propose not to open the link if the keyboard is shown, because it means I am editing.

* * *

* Android version: 12
* Android distribution: LineageOS
* App version: 1.5.0 F-Droid

**Comment by maltaisn** [comment]
I agree it's annoying. I'll look into it eventually...

**Comment by smaragdus** [comment]
I suppose that a good workaround for this problem would be implementation of **view mode**/**edit mode**. In **view mode** links are clickable, in **edit mode** they are not.

**Comment by Urbexed** [comment]
I agree it's quite annoying

**Comment by maltaisn** [comment]
Here's what I changed:

- Clicked at the start and at the end of a link won't do anything. This also fixes the case where we clicked in the end margin very far from a link and it still opened it.
- Clicking on a link will open a confirmation dialog. If cancel is clicked, nothing will happen and text editing can be done at the clicked position.

![image](https://github.com/maltaisn/another-notes-app/assets/33690959/11bed0a6-e869-4cc1-aca9-623e72d7725e)
